### PR TITLE
Toggle to watchlist

### PR DESCRIPTION
**Pull request by pink-bunny**
Do nothing with an error.
I am not sure about the watchlist success type and action location.

### CODE REVIEW

File: state/movie/operations/toggleToWatchlist.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { toggleToWatchlistSuccess } from '../../data/actions';
+
+const toggleToWatchlistLogic = createLogic({
+  type: TOGGLE_TO_WATCHLIST,
+
+  async process({ getState, httpClient, action }, dispatch, done) {
+    const { id, isMovieInWatchlist } = action;
+    const state = getState();
+    const sessionId = getSessionId(state);
+    const newStatus = !isMovieInWatchlist;
```
Comment (plzen) [inline state/movie/operations/toggleToWatchlist.js]: It's better to send implicit action from component what logic should do, so change this value in the component, please

File: state/movie/operations/toggleToWatchlist.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    try {
+      await httpClient.post(`/account/{account_id}/watchlist?session_id=${sessionId}`, {
+        media_type: 'movie', /* eslint-disable-line */
+        media_id: id, /* eslint-disable-line */
+        watchlist: newStatus
+      });
+
+      dispatch(toggleToWatchlistSuccess(id, newStatus));
+    } catch (error) {
+      const errorMessage = error.response.data.status_message;
```
Comment (plzen) [inline state/movie/operations/toggleToWatchlist.js]: For this case let do nothing if error happens 👍

File: src/pages/Movie/TopInfo/ToggleToWatchlist/container.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+class ToggleToWatchlistContainer extends React.Component {
+  changeWatchlistStatus = () => {
+    const {
+      id,
+      isMovieInWatchlist,
+      toggleToWatchlist
+    } = this.props;
+
+    toggleToWatchlist(id, isMovieInWatchlist);
```
Comment (plzen) [inline src/pages/Movie/TopInfo/ToggleToWatchlist/container.js]: You have action with signature `toggleToWatchlistSuccess = (id, newStatus) => {...}`, so it's better to send new status as described in method params. Just send opposite value.

File: src/pages/Movie/TopInfo/component.js [hunk 4]
```diff
@@ -107,23 +110,27 @@ const TopInfo = ({
...
+  id: PropTypes.number,
   overview: PropTypes.string,
   title: PropTypes.string,
   originalLanguage: PropTypes.string,
   runtime: PropTypes.number,
   budget: PropTypes.number,
   revenue: PropTypes.number,
-  genres: PropTypes.arrayOf(PropTypes.object)
+  genres: PropTypes.arrayOf(PropTypes.object),
+  isMovieInWatchlist: PropTypes.bool
```
Comment (plzen) [inline src/pages/Movie/TopInfo/component.js]: Is it correct that you might have movie without this value?

File: state/movie/operations/toggleToWatchlist.js (hunk 1 above)
Comment (pink-bunny) [reply state/movie/operations/toggleToWatchlist.js]: Fixed

File: state/movie/operations/toggleToWatchlist.js (hunk 2 above)
Comment (pink-bunny) [reply state/movie/operations/toggleToWatchlist.js]: Okey)

File: src/pages/Movie/TopInfo/ToggleToWatchlist/container.js (hunk 3 above)
Comment (pink-bunny) [reply src/pages/Movie/TopInfo/ToggleToWatchlist/container.js]: Fixed

File: src/pages/Movie/TopInfo/component.js (hunk 4 above)
Comment (pink-bunny) [reply src/pages/Movie/TopInfo/component.js]: It would be better to discuss it on the daily.

Comment (pink-bunny) [reply src/pages/Movie/TopInfo/component.js]: Fixed

File: state/data/actions.js [hunk 5]
```diff
@@ -9,3 +10,14 @@ export const mergeListsList = (lists) => ({
   type: types.MERGE_LISTS_LIST,
   lists
 });
+
+export const mergeMovie = (movie) => ({
```
Comment (plzen) [inline state/data/actions.js]: If I remember you already removed this function

File: state/data/types.js [hunk 6]
```diff
@@ -1,3 +1,5 @@
 export const MERGE_MOVIES_LIST = 'MERGE_MOVIES_LIST';
 
 export const MERGE_LISTS_LIST = 'MERGE_LISTS_LIST';
+
+export const MERGE_MOVIES = 'MERGE_MOVIES';
```
Comment (plzen) [inline state/data/types.js]: You don't need this action anymore

File: state/data/actions.js (hunk 5 above)
Comment (pink-bunny) [reply state/data/actions.js]: My fault. Fixed

File: state/data/types.js (hunk 6 above)
Comment (pink-bunny) [reply state/data/types.js]: Fixed

**Pull request merged**